mos-platform/c64/ldscripts/kernal.ld

```diff
@@ -0,0 +1,39 @@
+PROVIDE(CINT   = 0XFF81);
```

Comment:
What I was after here was a way to generate the call to kernal routine CHRIN such that clang/llvm would treat it as a normal function call, and thus generate `jsr __CHRIN`.  The only other thing I tried was the c-style solution of casting integer 0xFFCF to a function pointer of type `void(*)()`, If I did that, I always got an indirect call rather than a direct call.   Defining the routine as an explicit pre-defined symbol for the linker seemed to do the trick.  If there's a more elegant way of doing this I'm willing to learn.  I definitely don't know the best way of doing things in assembly.

Reply:
I think we're mostly on thee same page if you put the code snippet I gave in a "kernal.s" file, and include it in the link, then it's exactly equivalent to the linker script snippet you're using in terms of the behavior of the C and assembly code. It just uses the ordinary way of defining symbols: including object files in the link. The linker script mechanism is more intended for post-link hackery.